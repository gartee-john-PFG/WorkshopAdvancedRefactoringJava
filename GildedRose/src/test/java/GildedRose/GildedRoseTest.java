package GildedRose;


import org.junit.jupiter.api.Test;
import org.approvaltests.combinations.CombinationApprovals;

class GildedRoseTest {

    @Test
    void approveGildedRose() {
        String[] names = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Integer[] sellins = { -1, 0, 1, 5, 6, 10, 11, 12, 6, 2 };
        Integer[] qualities = { 0, 1, 48, 49, 50 };
        CombinationApprovals.verifyAllCombinations(
                this::runGildedRose,
                names,
                sellins,
                qualities);
    }

    private Item runGildedRose(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose gildedRose = new GildedRose(items);
        gildedRose.updateQuality();
        return gildedRose.items[0];
    }

}