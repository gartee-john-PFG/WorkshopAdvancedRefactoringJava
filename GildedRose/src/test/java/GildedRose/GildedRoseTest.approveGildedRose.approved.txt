[foo, -1, 0] => Item [name=foo, quality=0, sellIn=-2] 
[foo, -1, 1] => Item [name=foo, quality=0, sellIn=-2] 
[foo, -1, 48] => Item [name=foo, quality=46, sellIn=-2] 
[foo, -1, 49] => Item [name=foo, quality=47, sellIn=-2] 
[foo, -1, 50] => Item [name=foo, quality=48, sellIn=-2] 
[foo, 0, 0] => Item [name=foo, quality=0, sellIn=-1] 
[foo, 0, 1] => Item [name=foo, quality=0, sellIn=-1] 
[foo, 0, 48] => Item [name=foo, quality=46, sellIn=-1] 
[foo, 0, 49] => Item [name=foo, quality=47, sellIn=-1] 
[foo, 0, 50] => Item [name=foo, quality=48, sellIn=-1] 
[foo, 1, 0] => Item [name=foo, quality=0, sellIn=0] 
[foo, 1, 1] => Item [name=foo, quality=0, sellIn=0] 
[foo, 1, 48] => Item [name=foo, quality=47, sellIn=0] 
[foo, 1, 49] => Item [name=foo, quality=48, sellIn=0] 
[foo, 1, 50] => Item [name=foo, quality=49, sellIn=0] 
[foo, 5, 0] => Item [name=foo, quality=0, sellIn=4] 
[foo, 5, 1] => Item [name=foo, quality=0, sellIn=4] 
[foo, 5, 48] => Item [name=foo, quality=47, sellIn=4] 
[foo, 5, 49] => Item [name=foo, quality=48, sellIn=4] 
[foo, 5, 50] => Item [name=foo, quality=49, sellIn=4] 
[foo, 6, 0] => Item [name=foo, quality=0, sellIn=5] 
[foo, 6, 1] => Item [name=foo, quality=0, sellIn=5] 
[foo, 6, 48] => Item [name=foo, quality=47, sellIn=5] 
[foo, 6, 49] => Item [name=foo, quality=48, sellIn=5] 
[foo, 6, 50] => Item [name=foo, quality=49, sellIn=5] 
[foo, 10, 0] => Item [name=foo, quality=0, sellIn=9] 
[foo, 10, 1] => Item [name=foo, quality=0, sellIn=9] 
[foo, 10, 48] => Item [name=foo, quality=47, sellIn=9] 
[foo, 10, 49] => Item [name=foo, quality=48, sellIn=9] 
[foo, 10, 50] => Item [name=foo, quality=49, sellIn=9] 
[foo, 11, 0] => Item [name=foo, quality=0, sellIn=10] 
[foo, 11, 1] => Item [name=foo, quality=0, sellIn=10] 
[foo, 11, 48] => Item [name=foo, quality=47, sellIn=10] 
[foo, 11, 49] => Item [name=foo, quality=48, sellIn=10] 
[foo, 11, 50] => Item [name=foo, quality=49, sellIn=10] 
[foo, 12, 0] => Item [name=foo, quality=0, sellIn=11] 
[foo, 12, 1] => Item [name=foo, quality=0, sellIn=11] 
[foo, 12, 48] => Item [name=foo, quality=47, sellIn=11] 
[foo, 12, 49] => Item [name=foo, quality=48, sellIn=11] 
[foo, 12, 50] => Item [name=foo, quality=49, sellIn=11] 
[foo, 6, 0] => Item [name=foo, quality=0, sellIn=5] 
[foo, 6, 1] => Item [name=foo, quality=0, sellIn=5] 
[foo, 6, 48] => Item [name=foo, quality=47, sellIn=5] 
[foo, 6, 49] => Item [name=foo, quality=48, sellIn=5] 
[foo, 6, 50] => Item [name=foo, quality=49, sellIn=5] 
[foo, 2, 0] => Item [name=foo, quality=0, sellIn=1] 
[foo, 2, 1] => Item [name=foo, quality=0, sellIn=1] 
[foo, 2, 48] => Item [name=foo, quality=47, sellIn=1] 
[foo, 2, 49] => Item [name=foo, quality=48, sellIn=1] 
[foo, 2, 50] => Item [name=foo, quality=49, sellIn=1] 
[Aged Brie, -1, 0] => Item [name=Aged Brie, quality=2, sellIn=-2] 
[Aged Brie, -1, 1] => Item [name=Aged Brie, quality=3, sellIn=-2] 
[Aged Brie, -1, 48] => Item [name=Aged Brie, quality=50, sellIn=-2] 
[Aged Brie, -1, 49] => Item [name=Aged Brie, quality=50, sellIn=-2] 
[Aged Brie, -1, 50] => Item [name=Aged Brie, quality=50, sellIn=-2] 
[Aged Brie, 0, 0] => Item [name=Aged Brie, quality=2, sellIn=-1] 
[Aged Brie, 0, 1] => Item [name=Aged Brie, quality=3, sellIn=-1] 
[Aged Brie, 0, 48] => Item [name=Aged Brie, quality=50, sellIn=-1] 
[Aged Brie, 0, 49] => Item [name=Aged Brie, quality=50, sellIn=-1] 
[Aged Brie, 0, 50] => Item [name=Aged Brie, quality=50, sellIn=-1] 
[Aged Brie, 1, 0] => Item [name=Aged Brie, quality=1, sellIn=0] 
[Aged Brie, 1, 1] => Item [name=Aged Brie, quality=2, sellIn=0] 
[Aged Brie, 1, 48] => Item [name=Aged Brie, quality=49, sellIn=0] 
[Aged Brie, 1, 49] => Item [name=Aged Brie, quality=50, sellIn=0] 
[Aged Brie, 1, 50] => Item [name=Aged Brie, quality=50, sellIn=0] 
[Aged Brie, 5, 0] => Item [name=Aged Brie, quality=1, sellIn=4] 
[Aged Brie, 5, 1] => Item [name=Aged Brie, quality=2, sellIn=4] 
[Aged Brie, 5, 48] => Item [name=Aged Brie, quality=49, sellIn=4] 
[Aged Brie, 5, 49] => Item [name=Aged Brie, quality=50, sellIn=4] 
[Aged Brie, 5, 50] => Item [name=Aged Brie, quality=50, sellIn=4] 
[Aged Brie, 6, 0] => Item [name=Aged Brie, quality=1, sellIn=5] 
[Aged Brie, 6, 1] => Item [name=Aged Brie, quality=2, sellIn=5] 
[Aged Brie, 6, 48] => Item [name=Aged Brie, quality=49, sellIn=5] 
[Aged Brie, 6, 49] => Item [name=Aged Brie, quality=50, sellIn=5] 
[Aged Brie, 6, 50] => Item [name=Aged Brie, quality=50, sellIn=5] 
[Aged Brie, 10, 0] => Item [name=Aged Brie, quality=1, sellIn=9] 
[Aged Brie, 10, 1] => Item [name=Aged Brie, quality=2, sellIn=9] 
[Aged Brie, 10, 48] => Item [name=Aged Brie, quality=49, sellIn=9] 
[Aged Brie, 10, 49] => Item [name=Aged Brie, quality=50, sellIn=9] 
[Aged Brie, 10, 50] => Item [name=Aged Brie, quality=50, sellIn=9] 
[Aged Brie, 11, 0] => Item [name=Aged Brie, quality=1, sellIn=10] 
[Aged Brie, 11, 1] => Item [name=Aged Brie, quality=2, sellIn=10] 
[Aged Brie, 11, 48] => Item [name=Aged Brie, quality=49, sellIn=10] 
[Aged Brie, 11, 49] => Item [name=Aged Brie, quality=50, sellIn=10] 
[Aged Brie, 11, 50] => Item [name=Aged Brie, quality=50, sellIn=10] 
[Aged Brie, 12, 0] => Item [name=Aged Brie, quality=1, sellIn=11] 
[Aged Brie, 12, 1] => Item [name=Aged Brie, quality=2, sellIn=11] 
[Aged Brie, 12, 48] => Item [name=Aged Brie, quality=49, sellIn=11] 
[Aged Brie, 12, 49] => Item [name=Aged Brie, quality=50, sellIn=11] 
[Aged Brie, 12, 50] => Item [name=Aged Brie, quality=50, sellIn=11] 
[Aged Brie, 6, 0] => Item [name=Aged Brie, quality=1, sellIn=5] 
[Aged Brie, 6, 1] => Item [name=Aged Brie, quality=2, sellIn=5] 
[Aged Brie, 6, 48] => Item [name=Aged Brie, quality=49, sellIn=5] 
[Aged Brie, 6, 49] => Item [name=Aged Brie, quality=50, sellIn=5] 
[Aged Brie, 6, 50] => Item [name=Aged Brie, quality=50, sellIn=5] 
[Aged Brie, 2, 0] => Item [name=Aged Brie, quality=1, sellIn=1] 
[Aged Brie, 2, 1] => Item [name=Aged Brie, quality=2, sellIn=1] 
[Aged Brie, 2, 48] => Item [name=Aged Brie, quality=49, sellIn=1] 
[Aged Brie, 2, 49] => Item [name=Aged Brie, quality=50, sellIn=1] 
[Aged Brie, 2, 50] => Item [name=Aged Brie, quality=50, sellIn=1] 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-2] 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-2] 
[Backstage passes to a TAFKAL80ETC concert, -1, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-2] 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-2] 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-2] 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-1] 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-1] 
[Backstage passes to a TAFKAL80ETC concert, 0, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-1] 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-1] 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=0, sellIn=-1] 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=3, sellIn=0] 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=4, sellIn=0] 
[Backstage passes to a TAFKAL80ETC concert, 1, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=0] 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=0] 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=0] 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=3, sellIn=4] 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=4, sellIn=4] 
[Backstage passes to a TAFKAL80ETC concert, 5, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=4] 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=4] 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=4] 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=2, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=3, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=2, sellIn=9] 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=3, sellIn=9] 
[Backstage passes to a TAFKAL80ETC concert, 10, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=9] 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=9] 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=9] 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=1, sellIn=10] 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=2, sellIn=10] 
[Backstage passes to a TAFKAL80ETC concert, 11, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=49, sellIn=10] 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=10] 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=10] 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=1, sellIn=11] 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=2, sellIn=11] 
[Backstage passes to a TAFKAL80ETC concert, 12, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=49, sellIn=11] 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=11] 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=11] 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=2, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=3, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=5] 
[Backstage passes to a TAFKAL80ETC concert, 2, 0] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=3, sellIn=1] 
[Backstage passes to a TAFKAL80ETC concert, 2, 1] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=4, sellIn=1] 
[Backstage passes to a TAFKAL80ETC concert, 2, 48] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=1] 
[Backstage passes to a TAFKAL80ETC concert, 2, 49] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=1] 
[Backstage passes to a TAFKAL80ETC concert, 2, 50] => Item [name=Backstage passes to a TAFKAL80ETC concert, quality=50, sellIn=1] 
[Sulfuras, Hand of Ragnaros, -1, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=-1] 
[Sulfuras, Hand of Ragnaros, -1, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=-1] 
[Sulfuras, Hand of Ragnaros, -1, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=-1] 
[Sulfuras, Hand of Ragnaros, -1, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=-1] 
[Sulfuras, Hand of Ragnaros, -1, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=-1] 
[Sulfuras, Hand of Ragnaros, 0, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=0] 
[Sulfuras, Hand of Ragnaros, 0, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=0] 
[Sulfuras, Hand of Ragnaros, 0, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=0] 
[Sulfuras, Hand of Ragnaros, 0, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=0] 
[Sulfuras, Hand of Ragnaros, 0, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=0] 
[Sulfuras, Hand of Ragnaros, 1, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=1] 
[Sulfuras, Hand of Ragnaros, 1, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=1] 
[Sulfuras, Hand of Ragnaros, 1, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=1] 
[Sulfuras, Hand of Ragnaros, 1, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=1] 
[Sulfuras, Hand of Ragnaros, 1, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=1] 
[Sulfuras, Hand of Ragnaros, 5, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=5] 
[Sulfuras, Hand of Ragnaros, 5, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=5] 
[Sulfuras, Hand of Ragnaros, 5, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=5] 
[Sulfuras, Hand of Ragnaros, 5, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=5] 
[Sulfuras, Hand of Ragnaros, 5, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=5] 
[Sulfuras, Hand of Ragnaros, 6, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 10, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=10] 
[Sulfuras, Hand of Ragnaros, 10, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=10] 
[Sulfuras, Hand of Ragnaros, 10, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=10] 
[Sulfuras, Hand of Ragnaros, 10, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=10] 
[Sulfuras, Hand of Ragnaros, 10, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=10] 
[Sulfuras, Hand of Ragnaros, 11, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=11] 
[Sulfuras, Hand of Ragnaros, 11, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=11] 
[Sulfuras, Hand of Ragnaros, 11, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=11] 
[Sulfuras, Hand of Ragnaros, 11, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=11] 
[Sulfuras, Hand of Ragnaros, 11, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=11] 
[Sulfuras, Hand of Ragnaros, 12, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=12] 
[Sulfuras, Hand of Ragnaros, 12, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=12] 
[Sulfuras, Hand of Ragnaros, 12, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=12] 
[Sulfuras, Hand of Ragnaros, 12, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=12] 
[Sulfuras, Hand of Ragnaros, 12, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=12] 
[Sulfuras, Hand of Ragnaros, 6, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 6, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=6] 
[Sulfuras, Hand of Ragnaros, 2, 0] => Item [name=Sulfuras, Hand of Ragnaros, quality=0, sellIn=2] 
[Sulfuras, Hand of Ragnaros, 2, 1] => Item [name=Sulfuras, Hand of Ragnaros, quality=1, sellIn=2] 
[Sulfuras, Hand of Ragnaros, 2, 48] => Item [name=Sulfuras, Hand of Ragnaros, quality=48, sellIn=2] 
[Sulfuras, Hand of Ragnaros, 2, 49] => Item [name=Sulfuras, Hand of Ragnaros, quality=49, sellIn=2] 
[Sulfuras, Hand of Ragnaros, 2, 50] => Item [name=Sulfuras, Hand of Ragnaros, quality=50, sellIn=2] 
